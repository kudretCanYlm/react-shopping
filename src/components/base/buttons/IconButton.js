import PropTypes from 'prop-types';

const IconButton = ({ children, text, onClick = (f) => f, className = '' }) => {
  return (
    <div className={`btn-icon flex-row ${className}`} onClick={onClick}>
      {children}
      <p className="fontsize-dashboard white">{text}</p>
    </div>
  );
};

export default IconButton;

IconButton.propTypes = {
  children: PropTypes.object.isRequired,
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired
};
