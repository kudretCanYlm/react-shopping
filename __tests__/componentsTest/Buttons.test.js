import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import IconButton from 'components/base/buttons/IconButton';
import { BsAlignBottom } from 'react-icons/bs';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe('IconButtonTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(
        <IconButton onClick={() => {}} text="test">
          <BsAlignBottom fontSize={20} />
        </IconButton>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('subrenders correctly', () => {
    const subRender = <BsAlignBottom fontSize={20} />;

    const button = shallow(
      <IconButton onClick={() => {}} text="test">
        {subRender}
      </IconButton>
    );

    expect(button.contains(subRender)).toEqual(true);
  });

  it('onClick working', () => {
    //values
    let text = 'not clicked';
    const textAfter = 'clicked';

    //our onClick function
    const fun = () => {
      text = textAfter;
    };

    const component = renderer.create(
      <IconButton className="test-class" onClick={fun} text="test">
        <BsAlignBottom fontSize={20} />
      </IconButton>
    );

    let tree = component.toJSON();

    // manually trigger the callback
    renderer.act(() => {
      tree.props.onClick();
    });

    expect(text).toEqual(textAfter);
  });

  it('there is our className and text', () => {
    const className = 'test-className';
    const text = 'example text';

    const button = shallow(
      <IconButton onClick={() => {}} text={text} className={className}>
        <BsAlignBottom fontSize={20} />
      </IconButton>
    );

    expect(button.hasClass(className)).toEqual(true);
    expect(button.contains(text)).toEqual(true);
  });
});
