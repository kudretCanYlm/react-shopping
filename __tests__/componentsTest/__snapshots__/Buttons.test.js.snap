// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonTest render correctly 1`] = `
<div
  className="btn-icon flex-row "
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    fontSize={20}
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="12"
      rx="1"
      width="4"
      x="6"
      y="1"
    />
    <path
      d="M1.5 14a.5.5 0 0 0 0 1v-1zm13 1a.5.5 0 0 0 0-1v1zm-13 0h13v-1h-13v1z"
    />
  </svg>
  <p
    className="fontsize-dashboard white"
  >
    test
  </p>
</div>
`;
